managed/services/qan/client.go

```diff
@@ -180,7 +182,8 @@ func (c *Client) Collect(ctx context.Context, metricsBuckets []*agentv1.MetricsB
 
 	convertedMetricsBuckets := make([]*qanv1.MetricsBucket, 0, len(metricsBuckets))
 	for _, m := range metricsBuckets {
-		agent := agents[m.Common.AgentId]
+		agentID, _ := strings.CutPrefix(m.Common.AgentId, "/agent_id/")
+		agent := agents[agentID]
```

Comment:
is it really a case?
it should be QAN Agent ID, not pmm-agent ID, so this value should come to pmm-agent from the server

Reply:
Yeah, it believe it's legit. We'd better discuss.